beacon-chain/blockchain/new_slot.go

```diff
@@ -69,6 +68,12 @@ func (s *Service) NewSlot(ctx context.Context, slot types.Slot) error {
 				return err
 			}
 			s.store.SetJustifiedCheckptAndPayloadHash(bj, h)
```

Comment:
Could `UpdateJustifiedCheckpoint` be inside `SetJustifiedCheckptAndPayloadHash`. I can't think of why we would need one without another. This will reduce asymmetry and disparity in future implementation

Reply:
because `SetJustifiedCheckptAndPayloadHash`  does not have access to `s.cfg.ForkChoiceStore` .

Reply:
Annoying. Something to clean up later...

Reply:
Yeah, opened an issue that will make these calls obsolete. I assigned myself, will attack that this week